app/channel-settings/server/functions/saveReactWhenReadOnly.js

```diff
@@ -1,12 +1,21 @@
 import { Meteor } from 'meteor/meteor';
 import { Match } from 'meteor/check';
 
-import { Rooms } from '../../../models';
+import { Rooms, Messages } from '../../../models';
 
-export const saveReactWhenReadOnly = function(rid, allowReact) {
+export const saveReactWhenReadOnly = function(rid, allowReact, user, sendMessage = true) {
 	if (!Match.test(rid, String)) {
 		throw new Meteor.Error('invalid-room', 'Invalid room', { function: 'RocketChat.saveReactWhenReadOnly' });
 	}
 
-	return Rooms.setAllowReactingWhenReadOnlyById(rid, allowReact);
+	const result = Rooms.setAllowReactingWhenReadOnlyById(rid, allowReact);
+
+	if (result && sendMessage) {
+		if (allowReact) {
+			Messages.createRoomAllowedReactingByRoomIdAndUser(rid, user);
+		} else {
+			Messages.createRoomDisallowedReactingByRoomIdAndUser(rid, user);
```

Comment:
wouldn't be cleaner to use a ternary operator?  at least it would avoid nesting, but idk

```
allowReact ? Messages.createRoomAllowedReactingByRoomIdAndUser(rid, user)
	   : Messages.createRoomDisallowedReactingByRoomIdAndUser(rid, user);
```